modules/cli/src/main/scala/scala/cli/exportCmd/MavenProjectDescription.scala

```diff
@@ -0,0 +1,236 @@
+      ): Seq[MavenLibraryDependency] = {
+        val providedDeps = deps.toSeq.toList.map(_.value).map { dep =>
+          val org  = dep.organization
+          val name = dep.name
+          val ver  = dep.version
+          // TODO dep.userParams
+          // TODO dep.exclude
+          // TODO dep.attributes
+          val artNameWithPrefix = dep.nameAttributes match {
+            case NoAttributes           => name
+            case s: ScalaNameAttributes => s"${name}_$getScalaMajorPrefix"
+          }
+          val scope0 =
+            // FIXME This ignores the isCompileOnly when scope == Scope.Test
+            if (scope == Scope.Test) Some("test")
+            else if (isCompileOnly) Some("provided")
+            else None
+
+          MavenLibraryDependency(org, artNameWithPrefix, ver, scope0)
+        }
+
+        val scalaDep = if (!ProjectDescriptor.isPureJavaProject(options, sources)) {
+          // add scala dependency
+          // todo: get scala version from directive
+          val scalaDep = if scalaV.startsWith("3") then "scala3-library_3" else "scala-library"
+          List(MavenLibraryDependency("org.scala-lang", scalaDep, scalaV))
+        }
+        else Nil
+
+        providedDeps ++ scalaDep
+      }
+
+      toDependencies(options.classPathOptions.extraDependencies, true)
+    }
+
+    MavenProject(
+      dependencies = depSettings
+    )
+  }
+
+  private def getScalaVersion(options: BuildOptions): String =
+    options.scalaOptions.scalaVersion
+      .flatMap(_.versionOpt) // FIXME If versionOpt is empty, the project is pure Java
+      .getOrElse(ScalaCli.getDefaultScalaVersion)
+
+  private def plugins(
+    options: BuildOptions,
+    scope: Scope,
+    jdkVersion: String,
+    sourcesMain: Sources
+  ): MavenProject = {
+
+    // todo: use this method from mill and sbt projects as well
+    val pureJava = ProjectDescriptor.isPureJavaProject(options, sourcesMain)
+
+    val javacOptions = javacOptionsSettings(options)
+
+    val mavenJavaPlugin = buildJavaCompilerPlugin(javacOptions, jdkVersion)
+    val scalaPlugin     = buildScalaPlugin(javacOptions, jdkVersion, getScalaVersion(options))
+
+    val reqdPlugins = if (pureJava) Seq(mavenJavaPlugin) else Seq(scalaPlugin)
+
+    MavenProject(
+      plugins = reqdPlugins
+    )
+  }
+
+  private def buildScalaPlugin(
+    javacOptions: Seq[String],
+    jdkVersion: String,
+    scalaVersion: String
+  ): MavenPlugin = {
+
+    val compileMode = buildNode("recompileMode", "incremental")
+    val scalaVersionNode =
+      buildNode("scalaVersion", scalaVersion) // todo: set this value in properties
+    val javacOptionsElem = {
+      val opts = javacOptions.map { opt =>
+        buildNode("javacArg", opt)
+      }
+      <javacArgs>
+        {opts}
+      </javacArgs>
+    }
+
+    val configurationElements = Seq(compileMode, /*scalaVersionNone,*/ javacOptionsElem)
+
+    MavenPlugin(
+      "net.alchim31.maven",
+      "scala-maven-plugin",
+      "4.9.1",
```

Comment:
the version of the `scala-maven-plugin` should be configurable via a command line option, and the default should be exported to Mill's `deps.sc` and passed via `Constants` all the way here.